src/resources/prefabs.rs

```diff
@@ -1,48 +1,96 @@
 use std::collections::HashMap;
+use std::fs::read_dir;
 
 use amethyst::{
-    assets::{Handle, Prefab, PrefabLoader, RonFormat},
+    assets::{AssetStorage, Handle, Prefab, PrefabLoader, ProgressCounter, RonFormat},
     ecs::World,
+    utils::application_root_dir,
 };
 
-use crate::components::creatures::{CreaturePrefabData, CreatureType};
+use crate::components::creatures::CreaturePrefabData;
 
 #[derive(Default)]
 pub struct CreaturePrefabs {
-    prefabs: HashMap<CreatureType, Handle<Prefab<CreaturePrefabData>>>,
+    prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>,
 }
 
 impl CreaturePrefabs {
     pub fn insert(
         &mut self,
-        creature_type: CreatureType,
+        creature_type: String,
         prefab_handle: Handle<Prefab<CreaturePrefabData>>,
     ) {
         self.prefabs.insert(creature_type, prefab_handle);
     }
 
-    pub fn get_prefab(
-        &self,
-        creature_type: &CreatureType,
-    ) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
+    pub fn get_prefab(&self, creature_type: &str) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
         self.prefabs.get(creature_type)
     }
+
+    pub fn get_prefabs(&self) -> &HashMap<String, Handle<Prefab<CreaturePrefabData>>> {
+        &self.prefabs
+    }
+
+    pub fn set_prefabs(&mut self, prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>) {
+        self.prefabs = prefabs;
+    }
 }
 
 // Here we load all prefabs for the different creatures in the game.
 // These prefabs are then stored in a resource of type CreaturePrefabs that is used by the spawner system.
-pub fn initialize_prefabs(world: &mut World) {
+pub fn initialize_prefabs(world: &mut World) -> ProgressCounter {
     let mut creature_prefabs = CreaturePrefabs::default();
-    let (carnivore_prefab, herbivore_prefab, plant_prefab) =
-        world.exec(|loader: PrefabLoader<'_, CreaturePrefabData>| {
-            (
-                loader.load("prefabs/carnivore.ron", RonFormat, (), ()),
-                loader.load("prefabs/herbivore.ron", RonFormat, (), ()),
-                loader.load("prefabs/plant.ron", RonFormat, (), ()),
-            )
-        });
-    creature_prefabs.insert("Carnivore".to_string(), carnivore_prefab);
-    creature_prefabs.insert("Herbivore".to_string(), herbivore_prefab);
-    creature_prefabs.insert("Plant".to_string(), plant_prefab);
+    let mut progress_counter = ProgressCounter::new();
+    let prefab_iter = {
+        let prefab_dir_path = application_root_dir() + "/resources/prefabs";
```

Comment:
Please move the creatures into a separate subfolder, in case we add other prefabs